app/controllers/events_controller.rb

```diff
@@ -83,6 +88,80 @@ def schedule
     @schedule_type = @event.type.downcase!
   end
 
+  # GET /events/1/ranking
+  def ranking
+    # Array of RankingEntry Structs that gets sorted when filled completely
+    @ranking_entries = []
+
+    # TODO Either add the possibility to calculate rankings for single-player events
+    # TODO or replace `teams` with the to be introduced EventParticipant superclass
+    # Leaves the Array of RankingEntry Structs empty when no teams participate in the event
+    @event.teams.each do |team|
+      ranking_entry = RankingEntry.new(nil, team.name, 0, 0, 0, 0, 0, 0, 0, 0)
+
+      event_matches = @event.matches
+      # Considers only the team's home matches that belong to the event
+      home_matches_in_event = team.home_matches & event_matches
+      parse_matches_data_into_ranking_entry team, ranking_entry, home_matches_in_event, :parse_match_details_for_home
```

Comment:
We could think about using destructuring (as described e. g. [here](http://tony.pitluga.com/2011/08/08/destructuring-with-ruby.html)) instead of passing the object to be worked on (the `ranking_entry`). The could could look like this:

```ruby
added_match_count, added_won_count, added_draw_count, 
added_lost_count, added_goals, added_goals_against, 
added_points = get_ranking_entry_additions_from_match team, 
                                                      home_matches_in_event, 
                                                      :get_ranking_entry_additions_for_home

ranking_entry.match_count += added_match_count
ranking_entry.won_count += added_won_count
ranking_entry.draw_count += added_draw_count
ranking_entry.lost_count += added_lost_count
ranking_entry.goals += added_goals
ranking_entry.goals_against += added_goals_against
ranking_entry.points += added_points
```

However, while writing this comment I noticed that this approach would result in way more bloated and less readable code. So we should stick with the way of passing the `ranking_entry` object around that you introduced! :+1: